airflow/api_fastapi/core_api/datamodels/connections.py

```diff
@@ -90,8 +91,74 @@ class ConnectionBody(BaseModel):
     extra: str | None = Field(default=None)
 
 
-class ConnectionBulkBody(BaseModel):
-    """Connections Serializer for requests body."""
+class ConnectionBulkCreateAction(BaseModel):
+    """Bulk Create Variable serializer for request bodies."""
+
+    action: Literal["create"] = "create"
+    connections: list[ConnectionBody] = Field(..., description="A list of connections to be created.")
+    action_if_exists: Literal["skip", "overwrite", "fail"] = "fail"
+
+
+class ConnectionBulkUpdateAction(BaseModel):
+    """Bulk Update Connection serializer for request bodies."""
+
+    action: Literal["update"] = "update"
+    connections: list[ConnectionBody] = Field(..., description="A list of connections to be updated.")
+    action_if_not_exists: Literal["skip", "fail"] = "fail"
 
-    connections: list[ConnectionBody]
-    overwrite: bool | None = Field(default=False)
+
+class ConnectionBulkDeleteAction(BaseModel):
+    """Bulk Delete Connection serializer for request bodies."""
+
+    action: Literal["delete"] = "delete"
+    connection_ids: list[str] = Field(..., description="A list of connection IDs to be deleted.")
+    action_if_not_exists: Literal["skip", "fail"] = "fail"
+
+
+class ConnectionBulkBody(BaseModel):
+    """Request body for bulk Connection operations (create, update, delete)."""
+
+    actions: list[ConnectionBulkCreateAction | ConnectionBulkUpdateAction | ConnectionBulkDeleteAction] = (
+        Field(..., description="A list of Connection actions to perform.")
+    )
+
+
+class ConnectionBulkActionResponse(BaseModel):
+    """
+    Serializer for individual bulk action responses.
+
+    Represents the outcome of a single bulk operation (create, update, or delete).
+    The response includes a list of successful connection_ids and any errors encountered during the operation.
+    This structure helps users understand which key actions succeeded and which failed.
+    """
+
+    success: list[str] = Field(
+        default_factory=list, description="A list of connection_ids representing successful operations."
+    )
+    errors: list[dict[str, Any]] = Field(
+        default_factory=list,
+        description="A list of errors encountered during the operation, each containing details about the issue.",
+    )
```

Comment:
For instance this can almost be used for all resources (connection, variables etc..), we just need to update the `suecces` field description.